recipe/patch_yaml/anaconda-client.yaml

```diff
@@ -0,0 +1,76 @@
+# from this code
+# if record_name == "anaconda-client":
+#     # Bump minimum `requests` requirement of `anaconda-client` 1.11.0
+#     # https://github.com/conda-forge/anaconda-client-feedstock/pull/35
+#     if parse_version(record["version"]) == parse_version("1.11.0"):
+#         i = -1
+#         deps = record["depends"]
+#         with suppress(ValueError):
+#             i = deps.index("requests >=2.9.1")
+#         if i >= 0:
+#             deps[i] = "requests >=2.20.0"
+#     if record.get("timestamp", 0) <= 1684878992896:  # 2023-05-23
+#         # https://github.com/conda-forge/conda-forge-ci-setup-feedstock/issues/242
+#         # https://github.com/conda-forge/anaconda-client-feedstock/issues/40
+#         if any("urllib3" in dep for dep in record["depends"]):
+#             _replace_pin(
+#                 "urllib3 >=1.26.4",
+#                 "urllib3 >=1.26.4,<2.0.0a0",
+#                 record["depends"],
+#                 record,
+#             )
+#         else:
+#             # old versions depended on urllib3 via requests;
+#             # requests 2.30+ allows urllib3 2.x
+#             for lower_bound in (">=2.9.1", ">=2.0", ">=2.20.0"):
+#                 _replace_pin(
+#                     f"requests {lower_bound}",
+#                     f"requests {lower_bound},<2.30.0a0",
+#                     record["depends"],
+#                     record,
+#                 )
+#     # https://github.com/conda-forge/anaconda-client-feedstock/pull/44
+#     # https://github.com/Anaconda-Platform/anaconda-client/issues/678
+#     if (parse_version(record["version"]) == parse_version("1.12.0")) and record[
+#         "build_number"
+#     ] == 0:
+#         # Guard python-dateutil dependency with trailing space in "python "
+#         python_pinning = [
+#             x for x in record["depends"] if x.startswith("python ")
+#         ]
+#         for pinning in python_pinning:
+#             _replace_pin(pinning, "python >=3.8", record["depends"], record)
+if:
+  name: anaconda-client
+  version: 1.11.0
+then:
+  - replace_depends:
+      old: requests >=2.9.1
+      new: requests >=2.20.0
+---
+if:
+  name: anaconda-client
+  timestamp_le: 1684878992896
+  has_depends: urllib3*
+then:
+  - replace_depends:
+      old: urllib3 >=1.26.4
+      new: urllib3 >=1.26.4,<2.0.0a0
+---
+if:
+  name: anaconda-client
+  timestamp_le: 1684878992896
+  not_has_depends: urllib3*
+then:
+  - replace_depends:
+      old: requests >=2.*
+      new: ${old},<2.30.0a0
```

Comment:
This is pretty nifty!

IIUC, it first matches whatever dep matches `requests >=2.*` and then can reference & reinsert the result of that?